spark/src/main/java/org/apache/iceberg/spark/source/BaseDataReader.java

```diff
@@ -132,6 +142,15 @@ protected InputFile getInputFile(String location) {
     return inputFiles.get(location);
   }
 
+  protected Map<Integer, ?> constantsMap(FileScanTask task, Schema readSchema) {
+    if (readSchema.findField(MetadataColumns.PARTITION_COLUMN_ID) != null) {
+      StructType partitionType = Partitioning.partitionType(table);
+      return PartitionUtil.constantsMap(task, partitionType, BaseDataReader::convertConstant);
```

Comment:
I'd prefer just to call the method rather than trying to optimize by not adding the partition entry.

Reply:
I'd prefer calling this all the time too. However, this would mean we can no longer query tables with unknown transforms. We need to know all transforms to build a common partition type. That's why I don't set the partition column if it is not requested.

Any thoughts on this, @rdblue?

Reply:
Can we leave unknown transforms out of the partition type instead? We can just ignore them if they're unknown?

Reply:
I'd probably consider persisting the partition type in the metadata instead. It might be confusing to silently ignore a partition column.

Reply:
I think it would be fine for unknown partitions right now. It would unblock this without much risk.